From c2718d31efc7b8268d2935c7d7af3f23b2feb3f5 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:12 -0400
Subject: [usb] cdc-acm: do not update PM busy on read errors

Message-id: <1410887463-20674-481-git-send-email-dzickus@redhat.com>
Patchwork-id: 95596
O-Subject: [RHEL7 PATCH 480/591] USB: cdc-acm: do not update PM busy on read errors
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 4a8ee5059a241114c644350b6cb564c729a340fa
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:49 2014 +0200

    USB: cdc-acm: do not update PM busy on read errors

    There's no need to update the runtime PM last_busy field on read urb
    errors (e.g. when the urb is being killed on shutdown).

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-acm.c b/drivers/usb/class/cdc-acm.c
index f038f39..3c7cfac 100644
--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -416,13 +416,15 @@ static void acm_read_bulk_callback(struct urb *urb)
   dev_dbg(&acm->data->dev, "%s - disconnected\n", __func__);
   return;
  }
- usb_mark_last_busy(acm->dev);
 
  if (urb->status) {
   dev_dbg(&acm->data->dev, "%s - non-zero urb status: %d\n",
        __func__, urb->status);
   return;
  }
+
+ usb_mark_last_busy(acm->dev);
+
  acm_process_read_urb(acm, urb);
 
  /* throttle device if requested by tty */
-- 
1.7.1